src/Symfony/Component/Asset/PathPackage.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset;
+
+use Symfony\Component\Asset\VersionStrategy\VersionStrategyInterface;
+
+/**
+ * Package that adds a base path to asset URLs in addition to a version.
+ *
+ * In addition to the provided base path, this package also automatically
+ * prepends the current request base path if a Context is available to
+ * allow a website to be hosted easily under any given path under the Web
+ * Server root directory.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class PathPackage extends Package
```

Comment:
After reviewing this PR in full, decoration is probably not worth it. It would only complicate the code that the end-user needs to write, and the component is flexible enough with the version strategy. It shouldn't be needed to extend the `Package` classes (they could actually be final).